numpy/f2py/crackfortran.py

```diff
@@ -3082,12 +3128,9 @@ def param_parse(d, params):
     """
     if "(" in d:
         # this dimension expression is an array
-        dname = d[:d.find("(")]
-        ddims = d[d.find("(") + 1:d.rfind(")")]
```

Comment:
Do you know what you are getting here, i.e. what `d` is/can be?  Since the code here works, I would expect that this is pretty well formatted and the balancing might not be accounting for much here?

I think I am getting at the parentheses balancing seems a bit odd and I would like to know if there is even a reason for it.

(Also nit, but `replace_variables_in_equation` isn't recursive :).)